src/main/java/algorithms/curated170/hard/ClosestBinarySearchTreeValueIILinkedList.java

```diff
@@ -0,0 +1,39 @@
+package algorithms.curated170.hard;
+
+import algorithms.datastructures.TreeNode;
+import java.util.LinkedList;
+import java.util.List;
+
+public class ClosestBinarySearchTreeValueIILinkedList {
+
+    LinkedList<Integer> result;
+
+    public List<Integer> closestKValues(TreeNode root, double target, int k) {
+        result = new LinkedList<>();
+
+        if (root == null) {
+            return result;
+        }
+
+        inorder(root, target, k);
+
+        return result;
+    }
+
+    private void inorder(TreeNode curr, double target, int k) {
+        if (curr == null) {
+            return;
+        }
+
+        inorder(curr.left, target, k);
+        if (result.size() < k) {
+            result.add(curr.val);
+        } else if (Math.abs(curr.val - target) < Math.abs(result.getFirst() - target)) {
```

Comment:
As a note, we are comparing the current with the first because the first element is most probably the furthest from our target value. If it's not so, that means that the rest of the elements are further around some pivot inside the list, so we only get further away from this pivot when we do inorder travesal.